modules/docs/mdx/7.0-MIGRATION-GUIDE.mdx

```diff
@@ -538,6 +539,242 @@ main `react` package.
 - `Flex` was promoted from `@workday/canvas-kit-labs/layout` to `@workday/canvas-kit-react/layout`
 - `Stack` was promoted from `@workday/canvas-kit-labs/layout` to `@workday/canvas-kit-react/layout`
 
+## Model Changes
+
+### Guards and Callbacks
+
+We changed the signature of model event guards and callbacks. In v6, the parameters were in an
+object. This has a less than ideal developer experience as intellisense isn't engaged immediately
+and we don't plan on adding any additional parameters to guard and callback functions. We've removed
+the object wrapper.
+
+🤖 The codemod handles these changes automatically.
+
+Before:
+
+```tsx
+const model = useTabsModel({
+  onSelect({data: {id}, prevState}) {
+    console.log(id, prevState);
+  },
+});
+```
+
+After:
+
+```tsx
+const model = useTabsModel({
+  onSelect({id}, prevState) {
+    console.log(id, prevState);
+  },
+});
+```
+
+### Model Implementation
+
+If you don't extend models, you can skip this section.
+
+In v6 we supported Typescript 3.8 which limited the way types could be inferred and defined. Canvas
+Kit v7 now requires Typescript 4.1 which introduced
+[Template Literal Types](https://www.typescriptlang.org/docs/handbook/2/template-literal-types.html).
+This means that the guards and callbacks no longer need to be manually defined via an `eventMap`.
+Event maps were a stopgap and were manual and suffered from user-error in defining them. Not all
```

Comment:
Suggestions are not an option for some reason...
`Event maps were a stopgap and were manual and suffered from user-error in defining them.`
could be:
`Event maps were a stopgap, were manual, and suffered from user-error in defining them.`

Reply:
I had some thoughts about the language choices (`were manual`, `suffered from user-error`) but I think it's fine to let your personal tone shine through!

Reply:
"prone to errors" is the same idea without sounding like we're pointing fingers